File: seongwan/0826/BOJ_11723_집합.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	static StringBuilder sb = new StringBuilder();
	static int M, x;
	static Set<Integer> set = new HashSet<>();
	static String s;
	static boolean chk;

	public static void main(String[] args) throws Exception {
		M = Integer.parseInt(br.readLine());
		for (int i = 0; i < M; i++) {
			st = new StringTokenizer(br.readLine());
			s = st.nextToken();
			x = st.hasMoreTokens() ? Integer.parseInt(st.nextToken()) : 0;
			oper(s, x);
		}
		System.out.println(sb);

	}

	static void oper(String s, int x) {
		if (s.equals("add"))
			set.add(x);
		else if (s.equals("remove"))
			set.remove(x);
		else if (s.equals("check"))
			sb.append(set.contains(x) ? 1 : 0).append("\n");
		else if (s.equals("toggle"))
			if (set.contains(x))
				set.remove(x);
			else
				set.add(x);
		else if (s.equals("all"))
			for (int i = 1; i <= 20; i++) {
				set.add(i);
			}
		else if (s.equals("empty"))
			set.clear();
	}

}

File: woojae/2023_9/0924/BOJ_11723_집합.py
import sys

class main
    def __init__(self)
        self.m = int(input())
        self.s = set()

    def solve(self)
        for _ in range(self.m)
            i = sys.stdin.readline().rstrip().split()  # 명령 및 출력을 따로 받아서 처리하면 메모리 혹은 시간 초과 발생
            if i[0] == add
                self.s.add(int(i[1]))
            elif i[0] == remove
                self.s.discard(int(i[1]))
            elif i[0] == check
                print(1 if int(i[1]) in self.s else 0)
            elif i[0] == toggle
                if int(i[1]) in self.s
                    self.s.remove(int(i[1]))
                else
                    self.s.add(int(i[1]))
            elif i[0] == all
                self.s = set((range(1, 21)))
            else
                self.s.clear()

main = main().solve()